internal/applicationsnapshot/input.go

```diff
@@ -176,3 +210,79 @@ func readSnapshotSource(input []byte) (app.SnapshotSpec, error) {
 	log.Debugf("Read application snapshot from file %s", input)
 	return file, nil
 }
+
+func (*remoteClient) Get(ref name.Reference) (*remote.Descriptor, error) {
+	desc, err := remote.Get(ref)
+	if err != nil {
+		return nil, err
+	}
+	return desc, nil
+}
+
+func (*remoteClient) Index(ref name.Reference) (v1.ImageIndex, error) {
+	desc, err := remote.Index(ref)
+	if err != nil {
+		return nil, err
+	}
+	return desc, nil
+}
+
+func clientFromContext(ctx context.Context) (client, bool) {
+	cli, ok := ctx.Value(RemoteClientKey{}).(client)
+	return cli, ok
+}
+
+func expandImageIndex(ctx context.Context, snap *app.SnapshotSpec) error {
+	client, ok := clientFromContext(ctx)
+	if !ok {
+		client = &remoteClient{}
+	}
+	// For an image index, remove the original component and replace it with an expanded component with all its image manifests
+	var removeIndices []int
+	var modifiedComponents []app.SnapshotComponent
+	log.Debugf("Contents of snapComponents are %v", snap.Components)
+	for i, component := range snap.Components {
+		log.Debugf("Inside the loop. Container image is %v",component.ContainerImage)
+		ref, err := name.ParseReference(component.ContainerImage)
+		if err != nil {
+			return fmt.Errorf("unable to parse container image %s: %w", component.ContainerImage, err)
+		}
+
+		desc, err := client.Get(ref)
+		if err != nil {
+			return fmt.Errorf("unable to fetch descriptior for container image %s: %w. The client used was %T", ref, err, client)
+		}
+
+		if desc.MediaType.IsIndex() {
+			index, err := client.Index(ref)
+			if err != nil {
+				return fmt.Errorf("unable to fetch index for container image %s: %w", component.ContainerImage, err)
+			}
+
+			indexManifest, err := index.IndexManifest()
+			if err != nil {
+				return fmt.Errorf("unable to fetch image manifest for container image %s: %w", component.ContainerImage, err)
+			}
+
+			for _, manifest := range indexManifest.Manifests {
+				archComponent := component
+				archComponent.Name = fmt.Sprintf("%s-%s", component.Name, manifest.Platform.Architecture)
+				archComponent.ContainerImage = fmt.Sprintf("%s@%s", ref.Context().Name(), manifest.Digest)
```

Comment:
Should it be `ref.Repository.Name()` instead of `ref.Context().Name()`?

Reply:
huh, I thought Repository was the type. I'll check

Reply:
I think it should be `ref.Context().Name()` based on https://pkg.go.dev/github.com/google/go-containerregistry/pkg/name#Reference